sklearn/tree/tests/test_tree.py

```diff
@@ -208,18 +214,26 @@ def test_weighted_classification_toy():
                            "Failed with {0}".format(name))
 
 
-def test_regression_toy():
+@pytest.mark.parametrize("Tree", REG_TREES.values())
+@pytest.mark.parametrize("criterion", REG_CRITERIONS)
+def test_regression_toy(Tree, criterion):
     # Check regression on a toy dataset.
-    for name, Tree in REG_TREES.items():
-        reg = Tree(random_state=1)
-        reg.fit(X, y)
-        assert_almost_equal(reg.predict(T), true_result,
-                            err_msg="Failed with {0}".format(name))
+    if criterion == "poisson":
+        # make target positive
+        a = np.abs(np.min(y)) + 1
+        y_train = np.array(y) + a
```

Comment:
So here `y` and `true_results` are a list and making an array will make a copy.
However, this is not super explicit.

Could we trigger an explicit copy first of `y` and `true_results` with a comment?

Reply:
I added a comment. Is that enough?

I think you already know, but note that `np.array(..., copy=True)` is already the default. On top of that, the `+ a` creates another new array again.

Reply:
Yes it is enough.